rm_manual:
  robot_type: "hero"
  chassis:
    topic: "/controllers/chassis_controller/command"
    accel_x: 7
    accel_y: 10.0
    accel_z: 20
    safety_power: 50.0
    capacitor_threshold: 0.1
    charge_power: 20
    extra_power: 30
    burst_power: 200
  vel:
    topic: "/cmd_vel"
    max_linear_x: 4.0
    max_linear_y: 4.0
    max_angular_z: 10.0
    gyro_move_reduction: 0.25
  gimbal:
    topic: "/controllers/gimbal_controller/command"
    max_yaw_vel: 16
    max_pitch_vel: 8.37
    track_timeout: 0.5
    eject_sensitivity: 0.38
    target_cost_function:
      k_pos: 1.0
      k_vel: 0.0
      k_hp: 0.0
      k_freq: 99.0
      timeout: 0.1
  shooter:
    topic: "/controllers/shooter_controller/command"
    gimbal_error_limit: 0.05
    heat_limit:
      expect_shoot_frequency_1: 1
      expect_shoot_frequency_2: 2
      expect_shoot_frequency_3: 3
      burst_shoot_frequency: 5
      safe_shoot_frequency: 1
      heat_coeff: 3.0
      type: "ID1_42MM"
  cover:
    topic: "/controllers/cover_controller/command"
    on_pos: -2.0
    off_pos: 0.0
  detection_switch:
    fail_limit: 10

  controllers_list:
    state_controllers:
      - controllers/robot_state_controller
      - controllers/joint_state_controller
      - controllers/orientation_controller
    main_controllers:
      - controllers/chassis_controller
      - controllers/gimbal_controller
      - controllers/shooter_controller

  ui:
    trigger_change:
      - name: "chassis"
        config: { start_position: [ 400, 750 ], size: 15, width: 2, title: "chassis: " }
      - name: "target"
        config: { start_position: [ 400, 700 ], size: 15, width: 2, title: "target:  " }
      - name: "exposure"
        config: { start_position: [ 400, 650 ], size: 15, width: 2, title: "exposure:" }
    time_change:
      - name: "capacitor"
        config: { start_position: [ 900, 100 ], size: 15, width: 2, delay: 0.5 }
    fixed:
      - name: "4.5m_str"
        config: { type: "string", size: 10, width: 1, color: "yellow",
                  start_position: [ 1000, 375 ], content: "4.5" }
      - name: "5.3m_str"
        config: { type: "string", size: 10, width: 1, color: "yellow",
                  start_position: [ 1000, 393 ], content: "5.3m" }
      - name: "5.3m_line"
        config: { type: "line", width: 2, color: "orange",
                  start_position: [ 954, 391 ], end_position: [ 976, 391 ] }
      - name: "mid"
        config: { type: "line", width: 2, color: "yellow",
                  start_position: [ 960, 302 ], end_position: [ 960, 540 ] }
    flash:
      - name: "spin"
        config: { start_position: [ 900, 750 ], size: 15, width: 2,
                  color: "yellow", content: "please spin!!", delay: 1.2 }
      - name: "armor0"
        config: { type: "circle", width: 3, radius: 50, color: "yellow", delay: 0.75 }
      - name: "armor1"
        config: { type: "circle", width: 3, radius: 50, color: "yellow", delay: 0.75 }
      - name: "armor2"
        config: { type: "circle", width: 3, radius: 50, color: "yellow", delay: 0.75 }
      - name: "armor3"
        config: { type: "circle", width: 3, radius: 50, color: "yellow", delay: 0.75 }
